cpp/src/reductions/scan.cu

```diff
@@ -0,0 +1,163 @@
+#include <cudf/column/column_view.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/iterator.cuh>
+
+#include <rmm/rmm.h>
+#include <cudf/utilities/error.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <utilities/device_atomics.cuh>
+#include <cub/device/device_scan.cuh>
+#include <cudf/reduction.hpp>
+
+
+namespace cudf {
+namespace experimental {
+
+namespace detail {
+
+template <typename T, typename Op>
+struct Scan {
+    static
```

Comment:
Why is this static? `call` is not a good name for a function... Name should reflect what it does.

Reply:
I don't see why this is a struct in the first place. Just make these three functions into free functions...

Reply:
I adapted the old code. I will restructure this whole thing.